src/codec.rs

```diff
@@ -290,6 +290,28 @@ pub trait Encode {
 	fn using_encoded<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
 		f(&self.encode())
 	}
+
+	/// Calculates the encoded size, used when the encoded bytes are not interesting.
```

Comment:
```suggestion
	/// Calculates the encoded size.
	///
	/// Should be used when the encoded data isn't required.
	///
	/// # Note
	///
	/// This works by using a special [`Output`] that only tracks the size. So, there are no allocations inside the 
	/// output. However, this can not prevent allocations that some types are doing inside their own encoding. 
```